src/Oscoin/Consensus/Nakamoto.hs

```diff
@@ -131,11 +131,18 @@ mineNakamoto
        , Hashable c (BlockHeader c (Sealed c PoW))
        )
     => Telemetry.Tracer m
-    -> (forall tx. [Block c tx (Sealed c PoW)] -> Telemetry.Traced Difficulty)
+    -> (forall tx. [Block c tx (Sealed c PoW)] -> Difficulty)
     -> Miner c PoW m
 mineNakamoto probed difiFn getBlocks unsealedBlock = do
     blks  <- getBlocks difficultyBlocks
-    diffy <- probed (difiFn blks)
+    let prevDifficulty = blockTargetDifficulty
+                       . blockHeader
+                       . NonEmpty.head
+                       . NonEmpty.fromList
+                       $ blks
+    let diffy = difiFn blks
+    when (prevDifficulty /= diffy) $
```

Comment:
This is new actually, and makes sense - since we weren't checking that the difficulty changed.